src/riot-rs-boards/nrf52/build.rs

```diff
@@ -1,5 +1,61 @@
 use ld_memory::{Memory, MemorySection};
 
+enum Partition {
+    FullDevice,
+    Riotboot {
+        bootloader: u64,
+        slots: u64,
+        slot_used: u64,
+    },
+}
+
+impl Partition {
+    fn from_env() -> Self {
+        let slot_flags = [
+            std::env::var_os("CARGO_FEATURE_RIOTBOOT_SLOT_0").is_some(),
+            std::env::var_os("CARGO_FEATURE_RIOTBOOT_SLOT_1").is_some(),
+        ];
+
+        let slot = match slot_flags {
+            [false, false] => None,
+            [true, false] => Some(0),
+            [false, true] => Some(1),
+            _ => panic!("Conflicting riotboot-slot- flags."),
+        };
+
+        if std::env::var_os("CARGO_FEATURE_RIOTBOOT_PARTITIONING_16K_2SLOTS").is_some() {
+            Partition::Riotboot {
+                bootloader: 16 * 1024,
+                slots: 2,
+                slot_used: slot.expect("riotboot-partitioning set but no slot set"),
+            }
+        } else {
+            if slot.is_some() {
+                panic!("riotboot-slot selected but no riotboot-partitioning set");
+            }
+            // FIXME: We should make this explicit
+            Partition::FullDevice
+        }
+    }
+
+    /// Given the chip's total flash size, return flash start and end of the usable flash region in
+    /// the given partition
+    fn get_rom(&self, total_rom: u64) -> (u64, u64) {
```

Comment:
I think this is duplicating what ld-memory already provides (though via environment variables): https://github.com/kaspar030/ld-memory/blob/ee1fd420c9d0071b3b3bbcf4853a6dcf17e0b27a/lib/ld-memory/src/lib.rs#L154-L182.

There's `NRF52_FLASH_OFFSET` that offsets the whole flash area (set to bootloader size), `NRF52_FLASH_SLOTS` to set to the total number of slots, `NRF52_FLASH_SLOT` which when unset, disables the slotting, of when set to a slot number (starting at 0), calculates for that slot. There's then `NRF52_FLASH_SLOT_OFFSET` which adds another offset within the slot (set to 256 or hdr_len).

Shouldn't that cover it?